### PR TITLE
feat: predictive factories — midpoints, progressions, solar arcs (v6.0.0a36)

### PR DESCRIPTION
**Pull request by g-battaglia**
## Summary

- **MidpointFactory** — pairwise midpoints with 90° dial and aspect-to-midpoint detection
- **SecondaryProgressionFactory** — day-for-a-year progressed chart (composable `AstrologicalSubjectModel`), BCE support via Julian Day arithmetic
- **SolarArcFactory** — progressed Sun arc with directed-to-natal aspect contacts
- **Progression chart rendering** — full dual-wheel SVG pipeline with type-safe renderer

## Key design decisions

- All time arithmetic uses Julian Day (`swe.julday`/`swe.revjul`) — works for any year including BCE
- BCE path converts UT→LMT before `from_birth_data()` to match `_calculate_time_conversions_bce` convention
- `DOUBLE_CHART_TYPES` centralized in `charts_utils.py` as single source of truth
- `natal_subject.active_points` used as natal targets in solar arc (not hardcoded defaults)
- Custom ayanamsa (`USER` sidereal mode) persisted on model with Pydantic `model_validator`

## Test plan

- [x] 50 predictive factory tests (midpoint math, boundary cases, aspect detection, error paths, solar arc geometry, BCE progressions, numerical baselines)
- [x] 4 Progression SVG fixture tests (classic, modern, dark, table-grid)
- [x] 442 chart drawer + BCE date tests pass
- [x] `get_type_hints()` works on all new Pydantic models (FastAPI safe)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Predictive toolkit: midpoints, secondary progressions (day‑for‑a‑year) and solar‑arc directions; new "Progression" dual‑wheel (natal + progressed) chart with progression‑specific titles/labels and improved sizing/layout behavior.  
  * XML context exporters for solar‑arc and midpoint analyses; default predictive point set exposed.

* **Documentation**
  * README, changelog and release notes updated with predictive guides, Web API wording, examples, and clarified license guidance.

* **Tests**
  * Regression and SVG/XML well‑formedness tests for predictive factories, CE/BCE progression charts, and serializers.

* **Chores**
  * Package version bumped and release entry added.

### CODE REVIEW

**Comment by g-battaglia** [comment]
@coderabbitai full review

**Comment by g-battaglia** [comment]
@coderabbitai review

**Comment by g-battaglia** [comment]
@coderabbitai full review

**Comment by g-battaglia** [comment]
@coderabbitai review

**Comment by g-battaglia** [comment]
@coderabbitai review

The requested fixes have been pushed. Please run another review.

**Comment by g-battaglia** [comment]
@coderabbitai full review

File: kerykeion/secondary_progressions/secondary_progression_factory.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+            progressed_lmt_jd, swe.JUL_CAL
+        )
+        return AstrologicalSubjectFactory.from_birth_data(
+            year=year,
+            month=month,
+            day=day,
+            hour=hour,
+            minute=minute,
+            seconds=seconds,
+            **common_kwargs,
```
[the remark this answers is not in the document]
Comment (g-battaglia) [reply kerykeion/secondary_progressions/secondary_progression_factory.py:328]: This appears to be a false positive on the current branch: `AstrologicalSubjectFactory.from_birth_data()` supports `seconds` as a keyword-only parameter (`*, seconds: int = 0`), and the BCE progression tests exercise this path successfully.

**Comment by g-battaglia** [comment]
@coderabbitai full review

**Comment by g-battaglia** [comment]
@coderabbitai full review

**Comment by g-battaglia** [comment]
@coderabbitai review

The requested follow-up fixes have been pushed in d2732021. Please run another review.
